master/usage/assigning-ip-addresses-topology.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Assigning IP addresses based on topology
+---
+
+## Why this is useful
+
+{{site.prodname}} can be configured to use specific IP pools for different topological areas.
+For example, you may want workloads in a particular rack, zone, or region to receive addresses from the same IP pool.
+This may be desirable either to reduce the number of routes required in the network or to meet requirements imposed by an external firewall device or policy.
+This guide walks through an example of how to set this up.
+
+## How this works
+At a high level, this feature is operated by the setting node labels and then selecting those
+nodes via node selector on your IP pool resources. Note that there are a variety of other ways
+to specify IP address assignment behavior that can be found in the
+[cni-plugin configuration reference document]({{site.baseurl}}/{{page.version}}/reference/cni-plugin/configuration).
+Each approach has its own use-case but this particular one takes on the lowest priority
+compared to the others (i.e. Specifying IP pools through configuration has precendence
+over IP addresses through annotations which has precendence over this IP pool
+assignment through node selectors).
+
+## Prerequisites
+
+**{{site.prodname}} IPAM**
+
+This guide only applies if you are using {{site.prodname}} IPAM.
+
+**Labeled nodes**
+
+In order to assign IP pools to specific nodes, these nodes must be labeled. See the documentation for [calicoctl label]({{site.baseurl}}/{{page.version}}/reference/calicoctl/commands/label) and [kubectl label](https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes/#add-a-label-to-a-node) for more information on how to do this. We recommend using `kubectl label`.
```

Comment:
I think we should be more opinionated here (drop calicoctl label reference altogether), and maybe not even call this a requirement.

i.e., the guide below can say "Step 1 - label your nodes". WDYT?